### PR TITLE
add contact information block pattern

### LINKED ISSUE #378: Block Pattern: Contact Information
**Describe the block pattern you're suggesting. What blocks does it include?**

When I mocked up footer widgets for Twenty Twenty-One, I included a pattern where the left column was aligned left, the middle column was aligned center, and the right column was aligned right:

image

This isn't currently possible in footer widgets without adding custom code to a text or html widget. However, this would make a good block pattern.

-----

**What use cases do you envision for this block pattern?**

Adding some contact information to the bottom of your homepage, or above a contact form.

-----

**Mockups, Screenshots, or block HTML**

image

```
<!-- wp:columns {"align":"wide"} -->
<div class="wp-block-columns alignwide"><!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph -->
<p><a href="mailto:#">contact@tt1.org<br></a>123-456-7890</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph {"align":"center"} -->
<p class="has-text-align-center">123 Main Street<br>Cambridge, MA, 02139</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {"verticalAlignment":"center"} -->
<div class="wp-block-column is-vertically-aligned-center"><!-- wp:social-links {"align":"right","className":"is-style-twentytwentyone-social-icons-color"} -->
<ul class="wp-block-social-links alignright is-style-twentytwentyone-social-icons-color"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /-->

<!-- wp:social-link {"url":"https://www.facebook.com/WordPress/","service":"facebook"} /-->

<!-- wp:social-link {"url":"https://twitter.com/WordPress","service":"twitter"} /-->

<!-- wp:social-link {"service":"instagram"} /-->

<!-- wp:social-link {"service":"linkedin"} /-->

<!-- wp:social-link {"service":"youtube"} /-->

<!-- wp:social-link {"url":"https://www.youtube.com/wordpress","service":"youtube"} /--></ul>
<!-- /wp:social-links --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->
```

### PR DESCRIPTION
**Pull request by GuidoOffermans**
Fixes #378 

## Summary

Add the contact information block pattern.

## Test instructions

This PR can be tested by following these steps:
1. Start editing a page.
1. Click on the `add block` button.
1. Click on patterns and select the `contact information` pattern.
1. Match the result with the screenshot from https://github.com/WordPress/twentytwentyone/issues/378

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

File: inc/block-patterns.php [hunk 1]
```diff
@@ -106,4 +106,14 @@
...
...@tt1.org<br></a>123-456-7890</p><!-- /wp:paragraph --></div><!-- /wp:column --><!-- wp:column --><div class="wp-block-column"><!-- wp:paragraph {"align":"center"} --><p class="has-text-align-center">123 Main Street<br>Cambridge, MA, 02139</p><!-- /wp:paragraph --></div><!-- /wp:column --><!-- wp:column {"verticalAlignment":"center"} --><div class="wp-block-column is-vertically-aligned-center"><!-- wp:social-links {"align":"right","className":"is-style-twentytwentyone-social-icons-color"} --><ul class="wp-block-social-links alignright is-style-twentytwentyone-social-icons-color"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /--><!-- wp:social-link {"url":"https://www.facebook.com/WordPress/","service":"facebook"} /--><!-- wp:social-link {"url":"https://twitter.com/WordPress","service":"twitter"} /--><!-- wp:social-link {"service":"instagram"} /--><!-- wp:social-link {"service":"linkedin"} /--><!-- wp:social-link {"service":"youtube"} /--><!-- wp:social-link {"url":"https://www.youtube.com/wordpress","service":"youtube"} /--></ul><!-- /wp:social-links --></div><!-- /wp:column --></div><!-- /wp:columns --><!-- wp:paragraph --><p></p><!-- /wp:paragraph -->',
```
Comment (kishanjasani) [inline inc/block-patterns.php]: Make default string translatable and escape those too.

**Review by kishanjasani — CHANGES_REQUESTED** [review]

File: inc/block-patterns.php (hunk 1 above)
Comment (GuidoOffermans) [reply inc/block-patterns.php]: Good suggestion, I will add this!

**Comment by kjellr** [comment]
As @kishanjasani noted, please make the strings inside the pattern translatable. For example: 

```
<p class="has-text-align-center">123 Main Street<br>Cambridge, MA, 02139</p>
```

could be: 

```
<p class="has-text-align-center">' . wp_kses_post( __( '123 Main Street<br>Cambridge, MA, 02139', 'twentytwentyone' ) ) . '</p>
```

Aside from that this looks good. It looks a little weird on mobile, but I'm not sure there's anything we can do about that: 

Screen Shot 2020-10-12 at 12 37 37 PM

**Comment by melchoyce** [comment]
> Aside from that this looks good. It looks a little weird on mobile, but I'm not sure there's anything we can do about that

Agreed — it would be great if we could include some custom breakpoint styles here, so the three columns collapse down into one column sooner. I'm not opposed to adding custom classes to pattern containers, but I know you've felt iffy about that, @kjellr ?

**Comment by kjellr** [comment]
Yeah, it seems weird to me to add specific alignment class overrides for this one use case.

**Comment by melchoyce** [comment]
We've run into the issue a couple times, where we'd like to make some minor style changes — I wonder if that's something we should upstream to Gutenberg so it can be explored on a systems level?

**Comment by GuidoOffermans** [comment]
@kjellr Thanks for the concreet feedback, That bit should be fixed. The mobile alignment sadly exceeds my expertise.

**Comment by kjellr** [comment]
Thank you, this looks good. The only thing I'm not sure about is this part: 

```
<p><a href="mailto:#">' . wp_kses_post( __( 'contact@tt1.org<br></a>123-456-7890', 'twentytwentyone' ) ) . '</p>
```

Is it weird to put only the closing `</a>` tag inside the translate function? cc-ing @carolinan for a second opinion. 

Aside from that this is good to go. As @melchoyce noted above, I think it would make sense to wait until Gutenberg has better controls for responsive settings. (Yes, Mel — feel free to open a ticket there if you'd like).

**Comment by pattonwebz** [comment]
> Is it weird to put only the closing `</a>` tag inside the translate function? cc-ing @carolinan for a second opinion.

Yeah it is a little strange. Also `wp_kses_post()` is a pretty heavy function to use to veify this is a plain text string. `esc_html__()` might be preferable.

Try to keep all the html outside of the text and use a `sprintf()` to merge all the parts of the string.

File: inc/block-patterns.php [hunk 2]
```diff
@@ -101,10 +101,20 @@
...
...--></div><!-- /wp:column --><!-- wp:column --><div class="wp-block-column"><!-- wp:paragraph {"align":"center"} --><p class="has-text-align-center">' . wp_kses_post( __( '123 Main Street<br>Cambridge, MA, 02139', 'twentytwentyone' ) ) . '</p><!-- /wp:paragraph --></div><!-- /wp:column --><!-- wp:column {"verticalAlignment":"center"} --><div class="wp-block-column is-vertically-aligned-center"><!-- wp:social-links {"align":"right","className":"is-style-twentytwentyone-social-icons-color"} --><ul class="wp-block-social-links alignright is-style-twentytwentyone-social-icons-color"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /--><!-- wp:social-link {"url":"https://www.facebook.com/WordPress/","service":"facebook"} /--><!-- wp:social-link {"url":"https://twitter.com/WordPress","service":"twitter"} /--><!-- wp:social-link {"service":"instagram"} /--><!-- wp:social-link {"service":"linkedin"} /--><!-- wp:social-link {"service":"youtube"} /--><!-- wp:social-link {"url":"https://www.youtube.com/wordpress","service":"youtube"} /--></ul><!-- /wp:social-links --></div><!-- /wp:column --></div><!-- /wp:columns --><!-- wp:paragraph --><p></p><!-- /wp:paragraph -->',
```
Comment (carolinan) [inline inc/block-patterns.php]: The email address needs to be fake or there will be spam. Please use example@example.com.

Comment (aristath) [inline inc/block-patterns.php]:
```suggestion
			'content'     => '<!-- wp:columns {"align":"wide"} --><div class="wp-block-columns alignwide"><!-- wp:column --><div class="wp-block-column"><!-- wp:paragraph --><p><a href="mailto:#">' . esc_html__( 'example@example.com', 'twentytwentyone' ) . '<br></a>' . esc_html__( '123-456-7890', 'twentytwentyone' ) . '</p><!-- /wp:paragraph --></div><!-- /wp:column --><!-- wp:column --><div class="wp-block-column"><!-- wp:paragraph {"align":"center"} --><p class="has-text-align-center">' . esc_html__( '123 Main Street', 'twentytwentyone' ) . '<br>' . esc_html__( 'Cambridge, MA, 02139', 'twentytwentyone' ) . '</p><!-- /wp:paragraph --></div><!-- /wp:column --><!-- wp:column {"verticalAlignment":"center"} --><div class="wp-block-column is-vertically-aligned-center"><!-- wp:social-links {"align":"right","className":"is-style-twentytwentyone-social-icons-color"} --><ul class="wp-block-social-links alignright is-style-twentytwentyone-social-icons-color"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /--><!-- wp:social-link {"url":"https://www.facebook.com/WordPress/","service":"facebook"} /--><!-- wp:social-link {"url":"https://twitter.com/WordPress","service":"twitter"} /--><!-- wp:social-link {"service":"instagram"} /--><!-- wp:social-link {"service":"linkedin"} /--><!-- wp:social-link {"service":"youtube"} /--><!-- wp:social-link {"url":"https://www.youtube.com/wordpress","service":"youtube"} /--></ul><!-- /wp:social-links --></div><!-- /wp:column --></div><!-- /wp:columns --><!-- wp:paragraph --><p></p><!-- /wp:paragraph -->',
```

**Comment by GuidoOffermans** [comment]
@kjellr @pattonwebz @aristath @carolinan Thanks! I appreciate all the feedback :)

**Review by kjellr — APPROVED** [review]
This looks great. Thanks for your work on it!

**Pull request merged**